graphql/src/store/resolver.rs

```diff
@@ -246,6 +247,24 @@ where
         super::prefetch::run(ctx, selection_set, self.store.clone()).map(|value| Some(value))
     }
 
+    fn locate_block(&self, bc: &BlockConstraint) -> Result<BlockNumber, QueryExecutionError> {
+        match bc.block {
+            BlockLocator::Number(number) => Ok(number),
+            BlockLocator::Hash(hash) => self
+                .store
+                .block_number(&bc.subgraph, hash)
+                .map_err(|e| e.into())
+                .and_then(|number| {
+                    number.ok_or_else(|| {
+                        QueryExecutionError::ValueParseError(
```

Comment:
Is that a `ValueParseError` then or an invalid argument error?

Reply:
I was wondering the same thing when I wrote the code; `ValueParseError` gets turned into `Failed to decode 'block.hash' value: 'no block with that hash found'` which IMHO fits the bill. And it has the added charm that it doesn't require a `Position` which I don't have at that point.

Reply:
Ok, let's roll with that :)